test/rosdep_repo_check/alpine.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2021, Open Source Robotics Foundation
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#     * Neither the name of the Willow Garage, Inc. nor the names of its
+#       contributors may be used to endorse or promote products derived from
+#       this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import tarfile
+from io import TextIOWrapper
+
+from . import open_gz_url
+from . import PackageEntry
+from . import RepositoryCacheCollection
+
+
+def replace_tokens(string, os_code_name, os_arch):
+    """Replace tokens in the repository base URL."""
+    for key, value in {
+        '$basearch': os_arch,
+        '$releasever': os_code_name,
+    }.items():
+        string = string.replace(key, value)
+    return string
+
+
+def parse_apkindex(f):
+    # An example of an APKINDEX entry. Entries are divided by a blank line.
+
+    # V:2.1.1-r0
+    # A:x86_64
+    # S:6645
+    # I:28672
+    # T:RTP proxy (documentation)
+    # U:https://www.rtpproxy.org/
+    # L:BSD-2-CLause
+    # o:rtpproxy
+    # m:Natanael Copa <ncopa@alpinelinux.org>
+    # t:1602354892
+    # c:183e99f73bb1223768aa7231a836a3e98e94c03e
+    # i:docs rtpproxy=2.1.1-r0
+
+    fio = TextIOWrapper(f)
+    while True:
+        entry = {}
+        while (l := fio.readline()) != '\n':
+            k, v = l.strip().split(':', 1)
+            entry[k] = v
+
+        yield entry
```

Comment:
The `while True` needs to be terminated:
```suggestion
        if entry:
            yield entry
        else:
            break
```

Reply:
Infinite loops are rarely good.